### PR TITLE
#40: Add turbo and pnpm syncing support

### LINKED ISSUE #40: Add turbo + pnpm syncing
Similar to the headless-form addon, toucan core should add turbo + pnpm syncing due to the issue with dependenciesMeta injection.

- https://github.com/CrowdStrike/ember-headless-form/pull/52#discussion_r1113357130
- https://github.com/vercel/turbo/issues/2306#issuecomment-1433561563
- https://github.com/pnpm/pnpm/issues/4965#issuecomment-1287926075

Workflow benefits:

- make a change in toucan-core
- pnpm build
- see the update in the docs-app immediately

The problem at the moment is that to do above you have to do the following, which is a huge drag on DX/workflow:

- make a change in toucan-core
- pnpm build
- stop the running docs app
- run `pnpm i -f` in `docs-app`
- start the `docs-app`

Adopting what is done in headless-form will ensure the DX is a lot smoother.

### PR DESCRIPTION
**Pull request by ynotdraw**
## 🚀 Description

This PR adds `turbo` and `pnpm` syncing to match `ember-headless-form`.  This PR should align with what is in https://github.com/CrowdStrike/ember-headless-form/pull/52.

This cuts CI time by ~2mins, or in half!

Closes #40 

---

## 🔬 How to Test

No visual changes, all CI related

- Green build

---

## 📸 Images/Videos of Functionality

N/A

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -193,21 +187,9 @@ jobs:
...
-          persist-credentials: false
-      - uses: ./.github/actions/pnpm
-      - uses: ./.github/actions/download-built-package
       - uses: marocchino/sticky-pull-request-comment@v2
         with:
           message: |+
             ## Preview URLs
-            GH Env: ${{ needs.PublishDocstoCloudflarePages.outputs.env }}
-            docs: ${{ needs.PublishDocstoCloudflarePages.outputs.url }}
-#            api docs: ${{ needs.PublishDocstoCloudflarePages.outputs.url }}/api/modules.html
```
Comment (ynotdraw) [inline .github/workflows/ci.yml:213]: Went ahead and deleted this comment as we aren't doing these yet (but also no plan to soon™).

**Comment by nicolechung** [comment]
For local development, do we still have to set up `pnpm link @crowdstrike/ember-toucan-core` and `pnpm i` (without committing the related package.json changes?

```

Build Error (WaitForTrees)

@crowdstrike/ember-toucan-core/package.json lists ./dist/_app_/components/button.js in ember-addon.app-js, but that file does not exist


Stack Trace and Error Report: /var/folders/jc/81dkjn9s2sg5cpt_66jd8l640000gn/T/error.dump.c5cd0949f5a2005a23bb20913f0234cf.log
```

Without that I am still getting this error...

I do notice that the `docs-app` watches for changes now when I change something in `package/ember-toucan-core` which is exciting!

File: docs-app/package.json [hunk 2]
```diff
@@ -20,8 +20,10 @@
     "lint:js": "eslint . --cache",
     "lint:hbs:fix": "ember-template-lint . --fix --no-error-on-unmatched-pattern",
     "lint:js:fix": "eslint . --fix",
+    "lint:types": "glint",
     "start": "ember serve",
-    "test:ember": "ember test"
+    "test:ember": "ember test",
+    "_syncPnpm": "pnpm sync-pnpm"
```
[the remark this answers is not in the document]
Comment (ynotdraw) [reply docs-app/package.json:26]: Nope! You can still use pnpm linking locally as it works today!

**Comment by ynotdraw** [comment]
> For local development, do we still have to set up `pnpm link @crowdstrike/ember-toucan-core` and `pnpm i` (without committing the related package.json changes?

Yup! We will still need to do this for now


> ```
> 
> Build Error (WaitForTrees)
> 
> @crowdstrike/ember-toucan-core/package.json lists ./dist/_app_/components/button.js in ember-addon.app-js, but that file does not exist
> 
> 
> Stack Trace and Error Report: /var/folders/jc/81dkjn9s2sg5cpt_66jd8l640000gn/T/error.dump.c5cd0949f5a2005a23bb20913f0234cf.log
> ```
> 
> Without that I am still getting this error...
> 
> I do notice that the `docs-app` watches for changes now when I change something in `package/ember-toucan-core` which is exciting!

Interesting.  Can you provide some reproduction steps?  I tried the following with success:

```bash
rm -rf node_modules
pnpm i
cd docs-app/
pnpm start
```

and then from the root:

```bash
pnpm start:docs
```

I wonder if your node_modules were in a weird state or something?  May also try `rm -rf node_modules` from within the `docs-app` directory and re-`pnpm i`ing before starting?  Nothing should change workflow wise for us with these changes, so if this is failing we definitely need to fix!

**Comment by nicolechung** [comment]
> May also try `rm -rf node_modules` from within the `docs-app` directory and re-`pnpm i`ing before starting? Nothing should change workflow wise for us with these changes, so if this is failing we definitely need to fix!

I did a full `nuke` (rm -rf node_modules` from all packages and apps) before starting...let me try it again and make sure all the `node_modules` are cleared)...

But I didn't do `pnpm link`.

**Review by nicolechung — APPROVED** [review]

**Review by danwenzel — APPROVED** [review]

**Review by simonihmig — APPROVED** [review]
🎉

**Pull request merged**